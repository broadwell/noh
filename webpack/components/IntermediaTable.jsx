import React from "react";
import PropTypes from "prop-types";
import IntermediaTitle from "./IntermediaTitle";

const IntermediaTable = props => (
  // Should be refactored to generator automatically
  <div className="intermedia-table">
    <IntermediaTitle section={props.section} play={props.play} />

    <div className="intermedia__element">
      <div className="intermedia__label">Voices</div>
      <div className="intermedia__value">Waki/Waki Tsure</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Type of voice</div>
      <div className="intermedia__value">Sung - Non Congruent</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Text</div>
      <div className="intermedia__value">Non congruent - Sashinori</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Percussion</div>
      <div className="intermedia__value">Otsuzumi + Kotsuzumi + Taiko</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Type of percussion</div>
      <div className="intermedia__value">Non Congruent</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Nohkan</div>
      <div className="intermedia__value">Yes - Non congruent</div>
    </div>
    <div className="intermedia__element">
      <div className="intermedia__label">Dance</div>
      <div className="intermedia__value">Yes - Dance to text</div>
    </div>
  </div>
);

IntermediaTable.propTypes = {
  section: PropTypes.string.isRequired,
  play: PropTypes.string.isRequired
};

export default IntermediaTable;
