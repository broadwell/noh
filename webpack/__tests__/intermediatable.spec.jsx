import React from "react";
import { shallow } from "enzyme";
import IntermediaTable from "../components/IntermediaTable";

describe("<IntermediaTable>", () => {
  it("renders as expected", () => {
    const component = shallow(<IntermediaTable />);
    expect(component).toMatchSnapshot();
  });
});
