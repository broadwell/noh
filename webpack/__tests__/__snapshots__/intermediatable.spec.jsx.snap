// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InterMediaTable> renders as expected while connected 1`] = `
<IntermediaTable
  currentTime={10}
  dispatch={[Function]}
  play="hashitomi"
  sections={
    Array [
      Object {
        "dancePresent": Object {
          "value": "Dance to text",
        },
        "endTime": Object {
          "value": 180,
        },
        "intensity": Object {
          "number": "5",
        },
        "nokhanPresent": Object {
          "value": "Non Congruent",
        },
        "numberOfPercussion": Object {
          "value": "Otsuzumi + Kotsuzumi + Taiko",
        },
        "numberVoices": Object {
          "value": "Waki/Waki Tsure",
        },
        "percussion": Object {
          "value": "Non Congruent",
        },
        "sectionName": Object {
          "value": "Nanori",
        },
        "sectionUrl": "kokaji/nanori",
        "startTime": Object {
          "value": 0,
        },
        "text": Object {
          "value": "Sashinori - Non Congruent",
        },
        "voice": Object {
          "value": "Sung - Non Congruent",
        },
      },
      Object {
        "dancePresent": Object {
          "value": "Dance to text",
        },
        "endTime": Object {
          "value": 300,
        },
        "intensity": Object {
          "number": "5",
        },
        "nokhanPresent": Object {
          "value": "Non Congruent",
        },
        "numberOfPercussion": Object {
          "value": "Otsuzumi + Kotsuzumi + Taiko",
        },
        "numberVoices": Object {
          "value": "Waki/Waki Tsure",
        },
        "percussion": Object {
          "value": "Non Congruent",
        },
        "sectionName": Object {
          "value": "Mondo",
        },
        "sectionUrl": "kokaji/mondo",
        "startTime": Object {
          "value": 181,
        },
        "text": Object {
          "value": "Sashinori - Non Congruent",
        },
        "voice": Object {
          "value": "Sung - Non Congruent",
        },
      },
    ]
  }
/>
`;

exports[`<UnwrappedIntermediaTable> renders as expected 1`] = `
<div
  className="intermedia-table"
>
  <IntermediaTitle
    section="Nanori"
    sectionUrl="kokaji/nanori"
  />
  <IntermediaElement
    fieldName="Voices"
    fieldValue="Waki/Waki Tsure"
  />
  <IntermediaElement
    fieldName="Type of voice"
    fieldValue="Sung - Non Congruent"
  />
  <IntermediaElement
    fieldName="Text"
    fieldValue="Sashinori - Non Congruent"
  />
  <IntermediaElement
    fieldName="Percussion"
    fieldValue="Non Congruent"
  />
  <IntermediaElement
    fieldName="Type of percussion"
    fieldValue="Otsuzumi + Kotsuzumi + Taiko"
  />
  <IntermediaElement
    fieldName="Nohkan"
    fieldValue="Non Congruent"
  />
  <IntermediaElement
    fieldName="Dance"
    fieldValue="Dance to text"
  />
</div>
`;
